### PR TITLE
Generate code using controller-gen

### LINKED ISSUE #90: make generate-types doesn't work after moving to go modules
Since moving to go modules, generating types doesn't work:

```
kgz@sakura ~/g/s/g/s/c/cassandra-operator> make generate-types
== generate-types
/home/kgz/go/src/github.com/sky-uk/cassandra-operator/cassandra-operator/hack/update-codegen.sh
/home/kgz/go/src/github.com/sky-uk/cassandra-operator/cassandra-operator/hack/update-codegen.sh: line 12: /home/kgz/go/src/github.com/sky-uk/cassandra-operator/cassandra-operator/hack/../vendor/k8s.io/code-generator/generate-groups.sh: No such file or directory
make: *** [Makefile:143: generate-types] Error 127
```

We should run type generation in CI to avoid breaking it in the future.

### PR DESCRIPTION
**Pull request by kragniz**
Fixes #90.

### CODE REVIEW

**Comment by kragniz** [comment]
Depends on https://github.com/kubernetes-sigs/controller-tools/pull/222

File: cassandra-operator/hack/verify-codegen.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# based off https://github.com/jetstack/cert-manager/blob/master/hack/verify-crds.sh
```
Comment (sebbonnet) [inline cassandra-operator/hack/verify-codegen.sh]: wrong header :)

File: cassandra-operator/hack/verify-codegen.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+tmp="${output}/cassandra-operator"
+
+
+rsync -avvL "${REPO_ROOT}"/ "${tmp}" >/dev/null
+cd "${tmp}"
+"./hack/update-codegen.sh"
+
+echo "diffing against freshly generated codegen (${tmp})"
+ret=0
+diff -Naupr "${REPO_ROOT}/pkg/apis/cassandra/v1alpha1/zz_generated.deepcopy.go" "${tmp}/pkg/apis/cassandra/v1alpha1/zz_generated.deepcopy.go" || ret=$?
```
Comment (sebbonnet) [inline cassandra-operator/hack/verify-codegen.sh]: This looks very simiarl to `verify-crds.sh`, the only differences appear to be in the script invoked and the files to diff.
Is it worth refactoring this into a common script?

Comment (kragniz) [reply cassandra-operator/hack/verify-codegen.sh]: Right now they're pretty similar, but maybe in the future they could diverge? kubernetes/kubernetes also has a bunch of duplicated verify scripts https://github.com/kubernetes/kubernetes/tree/master/hack

**Comment by kragniz** [comment]
@sebbonnet yes, `zz_generated.deepcopy.go` in this branch is with my change in controller-tools

File: cassandra-operator/hack/verify-codegen.sh (hunk 2 above)
Comment (sebbonnet) [reply cassandra-operator/hack/verify-codegen.sh]: ok - lets leave it as is

**Pull request merged**